extensions/data-transfer/portability-data-transfer-generic/src/main/java/org/datatransferproject/datatransfer/generic/GenericTransferExtension.java

```diff
@@ -0,0 +1,173 @@
+package org.datatransferproject.datatransfer.generic;
+
+import static java.lang.String.format;
+import static org.datatransferproject.types.common.models.DataVertical.BLOBS;
+import static org.datatransferproject.types.common.models.DataVertical.SOCIAL_POSTS;
+
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+import java.util.stream.Collectors;
+import org.datatransferproject.api.launcher.ExtensionContext;
+import org.datatransferproject.spi.cloud.storage.JobStore;
+import org.datatransferproject.spi.transfer.extension.TransferExtension;
+import org.datatransferproject.spi.transfer.provider.Exporter;
+import org.datatransferproject.spi.transfer.provider.Importer;
+import org.datatransferproject.types.common.DownloadableItem;
+import org.datatransferproject.types.common.models.DataVertical;
+import org.datatransferproject.types.common.models.blob.BlobbyStorageContainerResource;
+import org.datatransferproject.types.common.models.blob.DigitalDocumentWrapper;
+import org.datatransferproject.types.common.models.social.SocialActivityContainerResource;
+
+class CachedDownloadableItem implements DownloadableItem {
+
+  private String cachedId;
+  private String name;
+
+  public CachedDownloadableItem(String cachedId, String name) {
+    this.cachedId = cachedId;
+    this.name = name;
+  }
+
+  @Override
+  public String getIdempotentId() {
+    return cachedId;
+  }
+
+  @Override
+  public String getFetchableUrl() {
+    // 'url' is ID when cached
+    return cachedId;
+  }
+
+  @Override
+  public boolean isInTempStore() {
+    return true;
+  }
+
+  @Override
+  public String getName() {
+    return name;
+  }
+}
+
+public class GenericTransferExtension implements TransferExtension {
+
+  private Map<DataVertical, Importer<?, ?>> importerMap = new HashMap<>();
+
+  class BlobbyContainerPath {
+    private BlobbyStorageContainerResource container;
+    private String path;
+
+    public BlobbyContainerPath(BlobbyStorageContainerResource container, String path) {
+      this.container = container;
+      this.path = path;
+    }
+
+    public BlobbyStorageContainerResource getContainer() {
+      return container;
+    }
+
+    public String getPath() {
+      return path;
+    }
+  }
+
+  @Override
+  public void initialize(ExtensionContext context) {
+    JobStore jobStore = context.getService(JobStore.class);
+
+    importerMap.put(
+        BLOBS,
+        new GenericFileImporter<BlobbyStorageContainerResource>(
+            (head, om) -> {
```

Comment:
Callbacks in-line are getting a bit unwieldy, so planning on refactoring this out to another file. That'll also help with making unit tests.